File: majority_element.py
from typing import List
def majority_element(a: List[int]) -> int:
    n = len(a)
    cnt = 0
    elem = 0
    for i in range(0, n):
        if cnt == 0:
            cnt = 1
            elem = a[i]
        elif elem == a[i]:
            cnt += 1
        else: cnt -= 1
    cnt = 0

    for i in range(n):
        if a[i] == elem:
            cnt += 1
    if cnt > (n // 2): return elem
    return -1

if __name__ == "__main__":
    print("The majority element is :")
    a = [2, 2, 1, 1, 1, 2, 2]
    print(majority_element(a))

File: majority_element.cpp
#include <iostream>
#include <vector>

using namespace std;


int majority_element(vector<int> a){
    int n = a.size();
    int el = -1;
    int cnt = 0;

    for (int i = 0; i < n; i++)
    {
        if(cnt == 0){
            cnt = 1;
            el = a[i];
        }
        else if(el == a[i]) cnt++;
        else cnt--;
    }

    cnt = 0;
    for (int i = 0; i < n; i++)
    {
        if(el == a[i]) cnt++;
    }
    if(cnt > (n / 2)) return el;
    return -1;
}

int main(){
    vector<int> arr = {2, 2, 1, 1, 1, 2, 2};
    int ans = majority_element(arr);
    cout << "The majority element is: " << ans << endl;
    return 0;
}